src/test/java/GmailServletTest.java

```diff
@@ -71,12 +69,79 @@
   private static final String MESSAGE_ID_THREE = "messageIdThree";
+  private static final String SENDER_ONE_NAME = "Sender_1";
+  private static final String SENDER_ONE_EMAIL = "senderOne@sender.com";
+  private static final String SENDER_TWO_EMAIL = "senderTwo@sender.com";
+  private static final String SENDER_WITH_NAME =
+      String.format("%s <%s>", SENDER_ONE_NAME, SENDER_ONE_EMAIL);
+  private static final String SENDER_WITHOUT_NAME = String.format("<%s>", SENDER_TWO_EMAIL);
+
+  private static final String DEFAULT_SENDER = "";
+  private static final int DEFAULT_N_DAYS = 7;
+  private static final int DEFAULT_M_HOURS = 3;
+  private static final int NEGATIVE_N_DAYS = -1;
+  private static final int NEGATIVE_M_HOURS = -1;
+
+  private static final String UNREAD_EMAIL_DAYS_QUERY =
+      String.format("newer_than:%dd is:unread ", DEFAULT_N_DAYS);
+  private static final String UNREAD_EMAIL_HOURS_QUERY =
+      String.format("newer_than:%dh is:unread ", DEFAULT_M_HOURS);
+  private static final String IMPORTANT_QUERY =
+      String.format("newer_than:%dd is:unread is:important ", DEFAULT_N_DAYS);
+
   private static final List<Message> NO_MESSAGES = ImmutableList.of();
-  private static final List<Message> THREE_MESSAGES =
+  private static final List<Message> THREE_MESSAGES_MAJORITY_SENDER_WITH_NAME =
       ImmutableList.of(
-          new Message().setId(MESSAGE_ID_ONE),
-          new Message().setId(MESSAGE_ID_TWO),
-          new Message().setId(MESSAGE_ID_THREE));
+          new Message()
+              .setId(MESSAGE_ID_ONE)
+              .setPayload(
+                  new MessagePart()
+                      .setHeaders(
+                          Collections.singletonList(
+                              new MessagePartHeader().setName("From").setValue(SENDER_WITH_NAME)))),
+          new Message()
+              .setId(MESSAGE_ID_TWO)
+              .setPayload(
+                  new MessagePart()
+                      .setHeaders(
+                          Collections.singletonList(
+                              new MessagePartHeader().setName("From").setValue(SENDER_WITH_NAME)))),
+          new Message()
+              .setId(MESSAGE_ID_THREE)
+              .setPayload(
+                  new MessagePart()
+                      .setHeaders(
+                          Collections.singletonList(
+                              new MessagePartHeader()
+                                  .setName("From")
+                                  .setValue(SENDER_WITHOUT_NAME)))));
+  private static final List<Message> THREE_MESSAGES_MAJORITY_SENDER_WITHOUT_NAME =
+      ImmutableList.of(
+          new Message()
+              .setId(MESSAGE_ID_ONE)
+              .setPayload(
+                  new MessagePart()
+                      .setHeaders(
+                          Collections.singletonList(
+                              new MessagePartHeader().setName("From").setValue(SENDER_WITH_NAME)))),
```

Comment:
Nit: Auxiliary method that takes an id and a name to avoid code repetition?

Reply:
Done!